File: programming-concepts/logical-operators/c-sharp/nor.cs
/*
Raspberry Pi Foundation
Developed as part of Ada Computer Science 
Usage licensed under CC BY-NC-SA 4.0

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;

namespace AdaCodeSamples {
    class LogicalOperators {
        // The Main method is the entry point for all C# programs
        public static void Main() {
            bool tired = false;
            bool earlyStart = false;

            if (!(tired || earlyStart)) {
                Console.WriteLine("I will go out tonight");
            }
            else {
                Console.WriteLine("I'm going to stay at home");
            }
        }
        

    }
}

File: programming-concepts/logical-operators/visual-basic/nor.vb
' Raspberry Pi Foundation
' Developed as part of Ada Computer Science
' Usage licensed under CC BY-NC-SA 4.0
' Note: This file is designed to be copied out and compiled on your machine.
' To run this file you need to:
' 1. Copy the contents
' 2. Paste them into the Visual Basic IDE of your choice (Visual Studio, for example)
' 3. Compile the program
' 4. Run the program

Module Program

    ' The Main subroutine is the default entry point for a VB program
    Sub Main()
        Dim tired As Boolean = False
        Dim earlyStart As Boolean = False

        If Not (tired OrElse earlyStart) Then
            Console.WriteLine("I will go out tonight")
        Else
            Console.WriteLine("I'm going to stay at home")
        End If
    End Sub

End Module
